webpack.config.ts

```diff
@@ -32,33 +36,69 @@ import { promisify } from 'util'
+    target: Target
+    mode: Configuration['mode']
+    dist: string
+    disableHMR?: boolean
+    disableReactHMR?: boolean
+    hmrPort?: number
+    noEval?: boolean
+}) {
+    const { disableReactHMR, mode, target, name, noEval, dist, hmrPort } = opts
+    let { disableHMR } = opts
     const isManifestV3 = target.runtimeEnv.manifest === 3
-    const enableHMR = env === 'development' && !Boolean(process.env.NO_HMR) && !isManifestV3
+    if (mode === 'production') disableHMR = true
+    if (mode === 'none') throw new TypeError('env cannot be none in this config')
 
-    /** On iOS, eval is async. */
-    const sourceMapKind: Configuration['devtool'] = target.iOS ? false : 'eval-source-map'
-    const ProcessEnvPlugin = new EnvironmentPlugin({ NODE_ENV: env, ...getGitInfo(), ...target.runtimeEnv })
+    /** On iOS, eval is async (it is hooked by webextension-shim). */
+    const sourceMapKind: Configuration['devtool'] = target.iOS || isManifestV3 || noEval ? false : 'eval-source-map'
     const config: Configuration = {
-        name: 'main',
-        mode: env,
-        devtool: env === 'development' ? sourceMapKind : false,
-        entry: {}, // ? Defined later
+        name,
+        mode,
+        devtool: mode === 'development' ? sourceMapKind : false,
+        target: ['web', 'es2018'],
+        experiments: { asset: true },
+        cache: {
+            type: 'filesystem',
+            buildDependencies: { config: [__filename] },
+            // In development mode we treat all envs as static. Each runtimeEnv will have it own cache. Therefor those modules won't be marked as uncacheable (and cause re-build very often).
+            // In production mode we mark them as runtime value so different targets can share a cache.
+            version: `1-node${process.version}-${mode === 'development' ? JSON.stringify(target.runtimeEnv) : 'build'}`,
+        },
         resolve: {
             extensions: ['.js', '.ts', '.tsx'],
-            //#region requirements of https://github.com/crimx/webpack-target-webextension
-            mainFields: ['browser', 'module', 'main'],
-            aliasFields: ['browser'],
-            //#endregion
-            alias: { 'async-call-rpc$': 'async-call-rpc/full', lodash: 'lodash-es' },
-
-            // If anyone need profiling React please checkout: https://github.com/facebook/create-react-app/blob/865ea05bc93fd2ac56b7e561181c7dc2cead3e78/packages/react-scripts/config/webpack.config.js#L304
+            alias: {
+                // If anyone need profiling React please checkout: https://github.com/facebook/create-react-app/blob/396892/packages/react-scripts/config/webpack.config.js#L338
+                'async-call-rpc$': 'async-call-rpc/full',
+                lodash: 'lodash-es',
+                // Strange...
+                '@dimensiondev/holoflows-kit': require.resolve('@dimensiondev/holoflows-kit/es'),
+                'xhr2-cookies': require.resolve('./scripts/package-overrides/xhr2-cookies'),
```

Comment:
Why we need this pkg?

Reply:
This is a indirect dependency of `web3-providers-http`. It depends on this `xhr2-cookies` and this package is re-implementing XHR on Node.JS (which we don't need it). I replaced this dependency because this `xhr2-cookies` imports so many Node-only packages I don't like to polyfill on.